### DISCUSSION TITLE
How to use system prompt without interactive mode?

### DISCUSSION THREAD

**Discussion by Naozumi520** [ggml-org/llama.cpp#2851] [Q&A]
I've seen few question in discussion about system prompt but no information about using system prompt with just a single question, just like the official openai

**Comment by ghost** [comment]
I dunno what openai's implementation is like. What's stopping you from using a system prompt with a single question?

I guess you're talking about llama2 chat models, yeah?

**Reply by Naozumi520** [reply]
Yeah. However, when I set the prompt, it always enter the interactive mode.

**Reply by ghost** [reply]
Show your `main`? ./main ect..

**Comment by shi-kejian** [comment]
```
<s>[INST] <<SYS>>
{{ system_prompt }}
<</SYS>>

{{ user_message }} [/INST]
```

According to https://huggingface.co/blog/llama2#how-to-prompt-llama-2

I guess you could just leave the {{system_prompt}} empty?

**Comment by goerch** [comment]
> I've seen few question in discussion about system prompt but no information about using system prompt with just a single question, just like the official openai

Yeah, `main` is pretty [complicated](https://github.com/ggerganov/llama.cpp/pull/3080). It would be nice to have some simple `text-completion` and `chat-completion` modes.
